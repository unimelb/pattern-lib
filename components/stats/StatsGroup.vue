<template>
  <div :class="['stats-group', { 'stats-group--inverted': inverted }]">
    <ul class="stats-group__list list-reset">
      <slot />
    </ul>
    <p
      v-if="attribution"
      class="stats-group__attribution">
      {{ attribution }}
    </p>
  </div>
</template>

<script>
export default {
  props: {
    inverted: {
      type: Boolean,
      default: false,
    },
    attribution: {
      type: String,
      default: '',
    },
  },
};
</script>
