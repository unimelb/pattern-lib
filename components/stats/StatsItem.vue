<template>
  <li class="stats-group__item">
    <!-- eslint-disable vue/no-v-html -->
    <div
      class="stats-group__icon"
      v-html="icon" />
    <!-- eslint-enable vue/no-v-html -->
    <strong class="stats-group__value">
      <span class="stats-group__value__hash">#</span>
      <span class="stats-group__value__num">{{ number }}</span>
    </strong>
    <span class="stats-group__text">{{ meta }}</span>
  </li>
</template>

<script>
export default {
  props: {
    number: {
      type: String,
      required: true,
    },
    meta: {
      type: String,
      required: true,
    },
    icon: {
      type: String,
      required: true,
    },
  },
};
</script>
