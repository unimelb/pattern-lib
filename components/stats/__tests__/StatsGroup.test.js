import {
  mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import StatsGroup from '../StatsGroup.vue';
import StatsItem from '../StatsItem.vue';

expect.extend(toHaveNoViolations);

describe('StatsGroup', () => {
  it('should match snapshot', () => {
    const result = mount(StatsGroup).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(StatsGroup, {
      slots: { StatsItem },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
