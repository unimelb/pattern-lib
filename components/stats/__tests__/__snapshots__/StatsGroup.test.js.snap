// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsGroup should match snapshot 1`] = `
<div
  class="stats-group"
>
  <ul
    class="stats-group__list list-reset"
  />
   
  <!---->
</div>
`;
